src/wp-includes/class-wp-script-modules.php

```diff
@@ -9,61 +9,62 @@
-	 *                                                                                 module version number. Defaults to
-	 *                                                                                 false. It is added to the URL as a
-	 *                                                                                 query string for cache busting
-	 *                                                                                 purposes. If $version is set to
-	 *                                                                                 false, the version number is the
-	 *                                                                                 currently installed WordPress
-	 *                                                                                 version. If $version is set to
-	 *                                                                                 null, no version is added.
+	 * @param string                                                        $id      The identifier of the script module.
+	 *                                                                               Should be unique. It will be used
+	 *                                                                               in the final import map.
+	 * @param string                                                        $src     Full URL of the script module, or
+	 *                                                                               path of the script module relative
+	 *                                                                               to the WordPress root directory.
+	 * @param array<string|array{id: string, import?: 'static'|'dynamic' }> $deps    Optional. An array of script module
+	 *                                                                               identifiers of the dependencies of
+	 *                                                                               this script module. The
+	 *                                                                               dependencies can be strings or
+	 *                                                                               arrays. If they are arrays, they
+	 *                                                                               need an `id` key with the script
+	 *                                                                               module identifier, and can contain
+	 *                                                                               an `import` key with either
+	 *                                                                               `static` or `dynamic`. By default,
+	 *                                                                               dependencies that don't contain an
+	 *                                                                               `import` key are considered static.
```

Comment:
See [conversation](https://wordpress.slack.com/archives/C02RQBWTW/p1705352387766709) in #core including [thread](https://wordpress.slack.com/archives/C02RQBWTW/p1705353393522159). The closest I think core's PHP documentation standards can come to describing this is:

```
@param array $deps {
    Dependencies.

    @type string|array $0... {
        An array of script module identifiers of the dependencies of
        this script module. The dependencies can be strings or arrays. 
        If they are arrays, they need an `id` key with the script module
        identifier, and can contain an `import` key with either `static`
        or `dynamic`. By default, dependencies that don't contain an
        `import` key are considered static.

        @type string $id     ID.
        @type string $import Import type. May be either 'static' or 'dynamic'. 
                             Optional. Defaults to 'static'.
    }
}
```

However, this would be exclusively for documentation and it would not provide any benefits for static analysis.

Reply:
I think that's fine. The `@type` syntax is used all over core, and from a documentation perspective I find it more useful and easier to parse than the current notation.

Reply:
Still, we should respect existing core patterns and not ad-hoc change them without further discussion.

Reply:
Dropped a note in #core-coding-standards: https://wordpress.slack.com/archives/C5VCTJGH3/p1705428874962969

Reply:
Also, @joedolson [suggested](https://wordpress.slack.com/archives/C5VCTJGH3/p1705431361320779) including both annotations.

Reply:
Ok, thanks. Would mind sharing an example of how to do that? I'm not familiar with the syntax.

Reply:
Or is it exactly [this syntax](https://github.com/WordPress/wordpress-develop/pull/5869#discussion_r1453781685)?

Reply:
Yes that's the syntax

Reply:
Thanks Pascal. I'll add the necessary changes to this PR.

Reply:
Changes applied in https://github.com/WordPress/wordpress-develop/pull/5869/commits/43631115bb71010a54287571229fa066dd237c71. 

Please let me know if that's the correct way to do so, as I don't have experience using this type of annotations yet.

Reply:
The best (only) examples I'm aware of is in the filesystem classes, for example:

https://github.com/WordPress/wordpress-develop/blob/0b7165e60d4c0d8c22c1fd736ed8a5335a8f4439/src/wp-admin/includes/class-wp-filesystem-base.php#L836-L859

Suggested that as a model: https://github.com/WordPress/wordpress-develop/pull/5869#pullrequestreview-1828057004

Nevertheless, PhpStorm doesn't like the syntax:

![image](https://github.com/WordPress/wordpress-develop/assets/134745/e951b84d-2219-46c8-afa7-5eeffacb3b9d)

But at least it is consistent with core.

Reply:
It would also be nice to still include `array<string|array{id: string, import?: 'static'|'dynamic' }>`. Suggestionj for that: https://github.com/WordPress/wordpress-develop/pull/5869#discussion_r1456588174